packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -82,54 +86,120 @@ import {
   QueryBuilderNavigationBlocker,
   type QueryBuilderState,
 } from '@finos/legend-query-builder';
+import { LegendQueryTelemetryHelper } from '../application/LegendQueryTelemetryHelper.js';
+import type { HelpMenuItemEditorRender } from '../stores/LegendQueryApplicationPlugin.js';
 
 const QueryExportDialogContent = observer(
   (props: { exportState: QueryExportState }) => {
     const { exportState } = props;
     const applicationStore = useApplicationStore();
-    const allowCreate = exportState.allowPersist;
+
+    const [isExistingQueryName, setIsExistingQueryName] = useState(false);
+
+    const saveNewQuery = exportState.allowPersist;
     const allowSave = exportState.allowPersist && exportState.allowUpdate;
     const create = applicationStore.guardUnhandledError(() =>
       exportState.persistQuery(true),
     );
-    const save = applicationStore.guardUnhandledError(() =>
+    const updateQuery = applicationStore.guardUnhandledError(() =>
       exportState.persistQuery(false),
     );
+    const renameExport = Boolean(exportState.renameExport);
 
     // name
     const nameInputRef = useRef<HTMLInputElement>(null);
-    const changeName: React.ChangeEventHandler<HTMLInputElement> = (event) =>
+
+    const debouncedLoadQueries = useMemo(
+      () =>
+        debounce((input: string): void => {
+          flowResult(
+            exportState.editorStore.queryLoaderState.loadQueries(input),
+          ).catch(applicationStore.alertUnhandledError);
+        }, 500),
+      [applicationStore, exportState.editorStore.queryLoaderState],
+    );
+
+    const changeName: React.ChangeEventHandler<HTMLInputElement> = (event) => {
       exportState.setQueryName(event.target.value);
+    };
 
     useEffect(() => {
       nameInputRef.current?.focus();
     }, []);
 
+    useEffect(() => {
+      const searchText = exportState.queryName.replace('Copy of ', '');
```

Comment:
can we not do these text hacks? What if we change the prefix tomorrow. Also, I think at this point we should get rid of the single quotes around query name when we say `copy of`, so e.g. we will generate name like `Copy of My Query`